server/src/controllers/experienceController.ts

```diff
@@ -0,0 +1,244 @@
+import { Request, Response, NextFunction} from "express";
+import { 
+    ExpListQuerySchema, 
+    ExpPutPostBodySchema,
+    ExpPatchBodySchema,
+    ExpPatchBody,
+    ExpPutPostBody,
+    ExpListQuery
+   } from "../models/experience";
+import { Prisma } from '../generated/prisma/client';
+import * as experienceService from "../services/experienceService";
+
+// --- CREATE ---
+
+import { AuthenticatedRequest } from "../middleware/authMiddleware";
+
+async function createExperience(
+    req: AuthenticatedRequest,
+    res: Response,
+    next: NextFunction
+    ) {
+    try {
+        const body: ExpPutPostBody = ExpPutPostBodySchema.parse(req.body);
+
+        if (!req.user) {
+            throw { status: 401, message: "Unauthorized" };
+        }
+
+        const experience = await experienceService.createExperience({
+            ...body,
+            createdBy: req.user.id,
+        });
+
+        return res.status(201).json(experience);
+    } catch (err) {
+        return next(err);
+    }
+}
+
+
+
+
+// --- READ ---
+
+
+// TODO: implement reviewCount, inTrips
+async function getExperience(
+    req: Request, 
+    res: Response,
+    next: NextFunction
+    ) {
+    try {
+        const experienceId = parseInt(req.params.id as string);
+
+        if (isNaN(experienceId) || experienceId <= 0) {
+            throw { status: 400, message: "Invalid experience ID"}
+        }
+
+        const experience = await experienceService.getExperience(experienceId);
+
+        if (!experience) {
+            throw { status: 404, message: "No experience with this id exists"}
+        }
+
+        return res.status(200).json(experience);
+    } catch (err) {
+        return next(err)
+    }
+}
+
+
+// TODO: implement filtering by tags, creator
+// TODO: add pagination info to response (nextOffset, prevOffset, etc)
+async function listExperiences(
+    req: Request, 
+    res: Response,
+    next: NextFunction
+    ) {
```

Comment:
Just a note -- you should install prettier. It will help your spacing, style and indentation be consistent. I'll be adding it soon